static/main.js

```diff
@@ -273,6 +274,26 @@ function initPolicies(options) {
     }
 }
 
+function filterComponentState(config, keysToRemove) {
+    function filterComponentStateImpl(component) {
+        if (component.content) {
+            for (var i = 0; i < component.content.length; i++) {
+                filterComponentStateImpl(component.content[i], keysToRemove);
+            }
+        }
+
+        if (component.componentState) {
+            Object.keys(component.componentState)
+                .filter(function (key) { return keysToRemove.includes(key); })
+                .forEach(function (key) { delete component.componentState[key]; });
+        }
+    }
+
+    config = cloneDeep(config);
```

Comment:
I'm wondering if this is needed, as toConfig() already makes new objects

Reply:
In my testing it was needed, toConfig seems to pass us references to some objects rather than copies?

I was a bit confused why I needed it as well having read the implementation of toConfig, so maybe I am wrong here.

Reply:
ah ok, can't be helped then